lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -2754,7 +2766,7 @@ void glow::fold(Function *F, CompilationMode mode) {
 
 void glow::optimize(Function *F, CompilationContext &cctx) {
   LOG_SCOPE(F->getLogContext(), "glow::optimize")
-
+  F->setState(FunctionState::FuncLoaded);
```

Comment:
Folding or no folding, isn't the deletion of constants a bug, please correct me if I am wrong?  Let's say when there are two functions coming from a same module, sharing the constants, deletion of constants by DCE should be automatically protected unless all the functions are completely loaded even if we are not doing the folding. My intent was to somehow indicate that a function is completely loaded in backward compatible way.  I thought compile is a safe enough place to assume that the function is completely loaded, and probably there is no harm even if it gets set multiple times. 

I have no objection to adding enableUnusedConstantDeletion, but this puts onus of setting this flag on the caller to ensure that all sister Functions are completely loaded. With functionState this was taken care of by the framework. Probably a binary flag isFuncLoaded would have been less confusing. Let me know if your prefer the optimizationOptions way, I will rollback the FunctionState and use your suggestion.

Reply:
I am currently creating a temporary function from the same module, so that I can reuse the constants. With the current implementation delete constants will happen only when for last function in the module is loaded, not before that. Is your suggested change going to break this implementation, because then we won't be able to compile the temporary function ?  Did I miss something in your suggestion?

Reply:
I think we will end up with a temporary Module instead of a temporary Function. So yeah this implementation would break. But we'd fix it when we make that change.